src/european_values/generative_training.py

```diff
@@ -68,39 +72,82 @@ def train_generative_model(
     n_components_range = [1, 2, 3, 4, 5, 10, 15, 20, 25, 30]
     n_components_range = [n for n in n_components_range if n <= max_components]
 
-    # Create GMM instance once
-    gmm = GaussianMixture(random_state=seed)
-
     bic_scores = {}
+
     for n_comp in n_components_range:
-        gmm.n_components = n_comp
-        gmm.fit(train_matrix)
-        bic_scores[n_comp] = gmm.bic(val_matrix)
+        # Create pipeline with scaler + GMM (as requested)
+        pipeline = make_pipeline(
+            MinMaxScaler(feature_range=(0, 1)),
+            GaussianMixture(n_components=n_comp, random_state=seed),
+        )
+
+        # Fit the entire pipeline on training data
+        pipeline.fit(train_matrix)
+        bic_scores[n_comp] = pipeline.named_steps["gaussianmixture"].bic(
+            pipeline.named_steps["minmaxscaler"].transform(val_matrix)
+        )
         logger.info(f"n_components={n_comp}: BIC={bic_scores[n_comp]:.2f}")
 
     best_n = min(bic_scores.keys(), key=lambda x: bic_scores[x])
     logger.info(f"Best n_components: {best_n}")
 
-    # Evaluate on test set - reuse the same GMM instance
-    logger.info("Evaluating on test set...")
-    gmm.n_components = best_n
-    gmm.fit(train_matrix)
-    test_bic = gmm.bic(test_matrix)
-    logger.info(f"Test BIC: {test_bic:.2f}")
+    # Create final pipeline with best number of components
+    final_pipeline = make_pipeline(
+        MinMaxScaler(feature_range=(0, 1)),
+        GaussianMixture(n_components=best_n, random_state=seed),
+    )
 
-    # Train final model on all data - reuse the same GMM instance
+    # Train final model on all data
     logger.info("Training final model on entire EU dataset...")
     full_matrix = eu_df[question_columns].values
-    gmm.fit(full_matrix)
+    final_pipeline.fit(full_matrix)
+
+    # Evaluate on test set
+    logger.info("Evaluating on test set...")
+    test_bic = final_pipeline.named_steps["gaussianmixture"].bic(
+        final_pipeline.named_steps["minmaxscaler"].transform(test_matrix)
+    )
+    logger.info(f"Test BIC: {test_bic:.2f}")
+
+    gmm = final_pipeline.named_steps["gaussianmixture"]
+    scaler = final_pipeline.named_steps["minmaxscaler"]
+    scaled_full_matrix = scaler.transform(full_matrix)
 
     logger.info(
-        f"Final model - BIC: {gmm.bic(full_matrix):.2f}, "
+        f"Final model - BIC: {gmm.bic(scaled_full_matrix):.2f}, "
         f"Converged: {gmm.converged_}, Iterations: {gmm.n_iter_}"
     )
 
-    # Save model
-    model_dir = Path("models")
-    model_dir.mkdir(exist_ok=True)
+    logger.info("Testing probabilities on training data...")
+    train_probabilities = gmm.predict_proba(scaled_full_matrix)
+    print(f"Training data - Component weights: {gmm.weights_}")
+    print(f"Training data - First 5 full probabilities:\n{train_probabilities[:5]}")
+
+    # Try weighted approach
+    weighted_probs = (
+        np.sum(train_probabilities * gmm.weights_, axis=1) / gmm.n_components
```

Comment:
Do we know that the multiplication is correct here? Since `gmm.weights_` is (I think) an array of shape `(n_components,)` and `train_probabilities` is of shape `(n_samples, n_components)`. We could, to be completely sure, replace `gmm.weights_` with `np.expand_dims(gmm.weights_, axis=0)`.

Also, thinking about it, we actually don't need to divide by number of components, since the weights sum up to 1.

```suggestion
        np.sum(train_probabilities * np.expand_dims(gmm.weights_, axis=0), axis=1)
```